IoTHubBuddy/IoTViewer/AccountManager.cs

```diff
@@ -0,0 +1,362 @@
+            WebAccount account = await WebAuthenticationCoreManager.FindAccountAsync(provider, accountId);
+            string clientId = "none";
+            if (authority == "organizations")
+            {
+                clientId = AzureActiveDirectoryClientId;
+                
+            }
+
+            WebTokenRequest request = new WebTokenRequest(provider, "wl.basic", clientId);
+            // Azure Active Directory requires a resource to return a token
+            request.Properties.Add("resource", "https://management.azure.com/");
+
+            WebTokenRequestResult result = await WebAuthenticationCoreManager.GetTokenSilentlyAsync(request, account);
+            if (result.ResponseStatus == WebTokenRequestStatus.UserInteractionRequired)
+            {
+                // Unable to get a token silently - you'll need to show the UI
+                return null;
+            }
+            else if (result.ResponseStatus == WebTokenRequestStatus.Success)
+            {
+                // Success
+                return result.ResponseData[0].Token;
+            }
+            else
+            {
+                // Other error 
+                return null;
+            }
+        }
+        /// <summary>
+        /// sign out of current account
+        /// </summary>
+        /// <returns></returns>
+        public static async Task SignOut()
+        {
+
+            string providerId = ApplicationData.Current.LocalSettings.Values["CurrentUserProviderId"]?.ToString();
+            string accountId = ApplicationData.Current.LocalSettings.Values["CurrentUserId"]?.ToString();
+
+            string authority = ApplicationData.Current.LocalSettings.Values["CurrentAuthority"]?.ToString();
+
+            if (null == providerId || null == accountId)
+            {
+                return;
+            }
+
+            WebAccountProvider provider = await WebAuthenticationCoreManager.FindAccountProviderAsync(providerId, authority);
+            WebAccount account = await WebAuthenticationCoreManager.FindAccountAsync(provider, accountId);
+            await SignOutAccountAsync(account);
+        }
+        /// <summary>
+        /// remove account from storage and sign out
+        /// </summary>
+        /// <param name="account"></param>
+        /// <returns></returns>
+        public static async Task SignOutAccountAsync(WebAccount account)
+        {
+            ApplicationData.Current.LocalSettings.Values.Remove("CurrentUserProviderId");
+            ApplicationData.Current.LocalSettings.Values.Remove("CurrentUserId");
+            ApplicationData.Current.LocalSettings.Values.Remove("CurrentAuthority");
+            ApplicationDataContainer accountsContainer = ApplicationData.Current.LocalSettings.Containers[AccountsContainer];
+            if(account.UserName=="")
+            {
+                accountsContainer.Containers[ProviderIdSubContainer].Values.Remove(account.Id);
```

Comment:
this is part of the reason why I want to change the storage layout. The consumer/organization accounts are set up differently so if account.UserName exists, then it is a consumer account and the account is stored with that. Otherwise, the account is stored by its id, and I need to retrieve it that way. I based my storage off the WAM sample (found [here](https://github.com/Microsoft/Windows-universal-samples/tree/master/Samples/WebAccountManagement)) but it's not ideal.